### PR TITLE
AddGallery

### PR DESCRIPTION
**Pull request by DavidAKopriva**
Add a Gallery page with plots.

### CODE REVIEW

**Comment by DavidAKopriva** [comment]
Ah, I hadn't thought about including videos of using the meshes.

**Comment by ranocha** [comment]
I was more thinking about showing the pictures of the meshes Andrew used for the videos. But we could also consider embedding the YouTube videos in the docs. Shall I try to do this?

**Comment by DavidAKopriva** [comment]
To be honest, I think linking the videos is a great idea. I tried again just now and the videos played without requiring a sign in. I don't know what was up with that, maybe the virtual vpn or something. A section with video links in the gallery would show that the meshes can be used.

**Comment by ranocha** [comment]
I copied the images to an external examples repository. We can delete them from this branch to keep this repository smaller.

**Comment by ranocha** [comment]
We should consider adding smaller versions of the images to the example repo - some are quite large and take quite a while to load.

**Comment by ranocha** [comment]
I have reduced the image sizes to decrease latency when loading the website. I have also added alternative formatting at the very end with a headline stating the title and contributor, followed by the picture of the mesh and, optionally a video. We could also consider adding more links there, e.g., to the control files etc.

Screenshot 2024-11-11 at 13 06 03

**Comment by DavidAKopriva** [comment]
My markdown editor is showing what I expect, so I am at a loss on why you are getting a single column and the comments are visible.
grab0

**Comment by DavidAKopriva** [comment]
I also just checked and exported the gallery to html, which, when displayed by safari, reproduces the picture above. It seems that the markdown that GitHub uses doesn't accept html commands? Or it ignores <div> since the two columns are not produced?

**Comment by DavidAKopriva** [comment]
Looking at Google's docs, https://github.github.com/gfm/#html-blocks, the html should be recognized and two columns should be produced.

**Comment by DavidAKopriva** [comment]
OK, I fixed that formatting issue. Still would like to know how to add captions.

**Comment by ranocha** [comment]
The markdown files are not served directly but are processed by `mkdocs`, see https://trixi-framework.github.io/HOHQMesh/building-the-documentation/.

File: Documentation/mkdocs.yml [hunk 1]
```diff
@@ -52,8 +53,10 @@ markdown_extensions:
...
   - pymdownx.superfences
 
-
 extra_javascript:
   - javascripts/config.js
   - https://polyfill.io/v3/polyfill.min.js?features=es6
   - https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js
+
+extra_css:
+  - css/video.css
```
Comment (ranocha) [inline Documentation/mkdocs.yml:62]: If you want to use new css, you can add a file here

File: Documentation/docs/css/video.css
Comment (ranocha) [inline Documentation/docs/css/video.css:1]: and add the file to this folder similar to this one. This should work and give you access to CSS in the generated html output.

**Comment by DavidAKopriva** [comment]
Thanks Michael, this is what I had in mind. I just didn't know how to hook up the css. I'm thinking of changing the format of the captions now that I see them. I think description (contributor) will look better.

**Comment by andrewwinters5000** [comment]
> Maybe the cylinder flow with curved boundaries from https://www.youtube.com/watch?v=w0A9X38cSe4 and the acoustic wave scattering mesh from https://www.youtube.com/watch?v=2v5KYdK7VLA could be added as well? CC @andrewwinters5000

I like the idea of adding video links as well. It shows the meshes "in action" and offers possibility for immediate additions to the gallery, e.g., the Malpasset test once we get the Shallow water package released.

**Comment by DavidAKopriva** [comment]
I think we have a good starting point for the gallery. I've edited the gallery page file to use the new div classes and edited the captions. Looks pretty good, imo. I edited the comments at the top of the file for the formats. Before merging the hohqmesh-gallery directory needs to be removed. Those file appear to be in the new GitHub repo with the examples.

**Comment by ranocha** [comment]
I removed the hohqmesh-gallery directory and added more links to videos.

**Review by ranocha — APPROVED** [review]

**Comment by sloede** [comment]
> Thanks Michael, this is what I had in mind. I just didn't know how to hook up the css.

No worries. And to be honest, I don't either - I actually had ChatGPT draw up all the required CSS for me (it seems to be quite HTML savvy... 😉)

**Comment by ranocha** [comment]
This PR is ready to be merged from my point of view 👍

**Review by andrewwinters5000 — APPROVED** [review]
Everything looks good from my end. Thanks so much for your help @ranocha !

**Pull request merged**